src/System.IO.Pipelines/ReadCursorOperations.cs

```diff
@@ -2,371 +2,94 @@
-                            index += _vectorSpan;
-
-                            if (block == end.Segment && index >= end.Index)
-                            {
-                                return -1;
-                            }
-
-                            continue;
-                        }
-
-                        if (block == end.Segment && index + byteIndex >= end.Index)
-                        {
-                            // Ensure iterator is left at limit position
-                            return -1;
-                        }
-
-                        result = new ReadCursor(block, index + byteIndex);
-                        return array.Array[array.Offset + index + byteIndex];
-                    }
-                    // Need unit tests to test Vector path
-#if !DEBUG
-                    }
-#endif
-
-                    fixed (byte* pCurrentFixed = array.Array)
-                    {
-                        var pCurrent = pCurrentFixed + array.Offset + index;
-
-                        var pEnd = block == end.Segment ? pCurrentFixed + array.Offset + end.Index : pCurrent + following;
-                        while (pCurrent < pEnd)
-                        {
-                            if (*pCurrent == byte0)
-                            {
-                                result = new ReadCursor(block, index);
-                                return byte0;
-                            }
-                            if (*pCurrent == byte1)
-                            {
-                                result = new ReadCursor(block, index);
-                                return byte1;
-                            }
-                            if (*pCurrent == byte2)
-                            {
-                                result = new ReadCursor(block, index);
-                                return byte2;
-                            }
-                            pCurrent++;
-                            index++;
-                        }
-                    }
-                    following = 0;
-                    break;
+                    result = new ReadCursor(segment, enumerator.CurrentSegmentStartIndex + index);
+                    return span[index];
                 }
             }
+
+            result = end;
+            return -1;
         }
 
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static Vector<byte> GetVector(byte vectorByte)
+        private static int MinIndex(int v1, int v2)
         {
-            // Vector<byte> .ctor doesn't become an intrinsic due to detection issue
-            // However this does cause it to become an intrinsic (with additional multiply and reg->reg copy)
-            // https://github.com/dotnet/coreclr/issues/7459#issuecomment-253965670
-            return Vector.AsVectorByte(new Vector<uint>(vectorByte * 0x01010101u));
+            v1 = v1 == -1 ? int.MaxValue : v1;
```

Comment:
So our searches return -1 for not found and then this routine is required to find the minimun index. If we had a IndexesOf method it would return the minimum index found in the first slot of the results span.

cc: @shiftylogic